Src/BootCamp.Chapter1/ArrayOperations.cs

```diff
@@ -63,8 +108,22 @@ public static int[] RemoveAt(int[] array, int index)
         /// <returns>A new array with element added at a given index. If an array is empty or null, returns new array with number in it.</returns>
         public static int[] InsertFirst(int[] array, int number)
         {
-            // ToDo: implement.
-            return array;
+            // Returns new number array if array is empty of null.
+            if (array == null || array.Length == 0) return new int[1] { number };
+
+            // Create new array with an extra slot
+            int[] new_array = new int[array.Length + 1];
+
+            // If at first position, add the number, else add value from old array - 1
+            for (int i = 0; i < new_array.Length; i++)
+            {
+                if (i == 0)
```

Comment:
Is there a way to write this that doesn't require a branch comparison of `i` to `0` each iteration of the loop?